docs/design/generics/details.md

```diff
@@ -3713,6 +3760,10 @@ class Array(T:! Type, template N:! Int) {
 }
 ```
 
+The `impl` declarations in the body of a `class` definition must be for the
```

Comment:
Does this apply only to internal `impl`s, or also to `external impl` declarations?

Reply:
I would say both, but I'm wondering if you have a use case for `external impl` of some other type? The only case I can think of it being useful is if you have an interface that requires `impl T as ComparableWith(Self)` you might want to do that external impl there.

Reply:
In person we discussed this and came to the conclusion that there are use cases for `external impl`s that should have access to private members, such as conversion from <some other type> to `Self`, or implementing <some other type> `+ Self`.